app/services/finance/extract_report.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Finance
+  class ExtractReport
+    def npq
+      raw_connection.copy_data(npq_sql) do
+        while (row = raw_connection.get_copy_data)
+          puts row.force_encoding("UTF-8")
+        end
+      end
+    end
+
+  private
+
+    def connection
+      @connection ||= ApplicationRecord.connection
+    end
+
+    def raw_connection
+      @raw_connection ||= connection.raw_connection
+    end
+
+    def npq_sql
+      <<-SQL
+        COPY (SELECT
+              clp.name                                               AS lead_provider_name,
+                na.school_urn                                        AS school_urn,
+                na.eligible_for_funding                              AS eligible_for_funding,
+                (SELECT name FROM schools WHERE urn = na.school_urn) AS school_name,
```

Comment:
Any reason to do a subquery here rather than `join schools`?

I _think_ this might result in a separate query per row (unconfirmed, not sure how PostgreSQL will deal with it internally).

Reply:
- It's copy paste job so didn't look too much into it, as it seems to work
- if you do a join and it doesn't return anything the row will be empty. 
- here the subquery will either return a string or null
- but more importantly if it is null it will still return the row unlike the join

Reply:
you could probably do so some kind of outer join to yield the same result

Reply:
Yeah, `left outer` will do the trick.